### PR TITLE
ref(backup): Fix mypy errors in backup.py

### PR DESCRIPTION
**Pull request by azaslavsky**
These changes make mypy happy with backup.py.

### CODE REVIEW

File: src/sentry/runner/commands/backup.py [hunk 1]
```diff
@@ -299,23 +298,23 @@ def sort_dependencies():
...
-                        deps.append(rel_model)
+                    deps.append(rel_model)
 
             # Also add a dependency for any simple M2M relation with a model
             # that defines a natural key.  M2M relations with explicit through
             # models don't count as dependencies.
-            for field in model._meta.many_to_many:
-                rel_model = field.remote_field.model
-                if rel_model != model:
+            meta: Any = model._meta
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/backup.py:314]: this should type fine -- what error are you getting here? (downcasting to `Any` is a difficult-to-find hack for when we want to clean this up later -- it's much more preferable to use `# type: ignore` with a link to an upstream mypy / django-stubs bug)

Comment (azaslavsky) [reply src/sentry/runner/commands/backup.py]: I was getting the following mypy error:

```
src/sentry/runner/commands/backup.py:315: error: "Options[Any]" has no attribute "many_to_many"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

When digging around, I discovered that we're using the internal-only Django `many_to_many` API here (see: `.../site-packages/django/db/models/options.py`, L527) anyway. I've changed it to filter the fields as the docstring at that line suggests instead, and attached a more specific type annotation to the variable declaration.

If this is not a good solution, I can just pop `# type: ignore` on the original code like you suggest.

File: src/sentry/runner/commands/backup.py [hunk 2]
```diff
@@ -299,23 +299,25 @@ def sort_dependencies():
...
-                        deps.append(rel_model)
+                    deps.append(rel_model)
 
             # Also add a dependency for any simple M2M relation with a model
             # that defines a natural key.  M2M relations with explicit through
             # models don't count as dependencies.
-            for field in model._meta.many_to_many:
-                rel_model = field.remote_field.model
-                if rel_model != model:
+            many_to_many_fields: list[ManyToManyField] = [
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/backup.py:315]: the annotation shouldn't be needed here because the `if` is narrowing

**Review by asottile-sentry — APPROVED** [review]

File: src/sentry/runner/commands/backup.py (hunk 2 above)
Comment (azaslavsky) [reply src/sentry/runner/commands/backup.py]: Done.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**